src/auth/auth.ts

```diff
@@ -778,4 +778,95 @@ export class Auth extends BaseAuth implements FirebaseServiceInterface {
     }
     return this.tenantsMap[tenantId];
   }
+
+  /**
+   * Looks up the tenant identified by the provided tenant ID and returns a promise that is
+   * fulfilled with the corresponding tenant if it is found.
+   *
+   * @param {string} tenantId The tenant ID of the tenant to look up.
+   * @return {Promise<Tenant>} A promise that resolves with the corresponding tenant.
+   */
+  public getTenant(tenantId: string): Promise<Tenant> {
+    return (this.authRequestHandler as AuthRequestHandler).getTenant(tenantId)
```

Comment:
You can use generics and get rid of these casts if you want to.

```
abstract class AbstractRequestHandler { }

class AuthRequestHandler extends AbstractRequestHandler {
  public getTenant() { }
}

class TenantAwareRequestHandler extends AbstractRequestHandler { }

abstract class BaseAuth<T extends AbstractRequestHandler> {
  protected requestHandler: T;
}

class Auth extends BaseAuth<AuthRequestHandler> {
  public getTenant() {
    // requestHandler is treated as AuthRequestHandler by the compiler
    this.requestHandler.getTenant();
  }
}
```

Reply:
Good idea. Done.